### PR TITLE
Added header for driver version and other info

### PR DESCRIPTION
**Pull request by tjoubert**
@joowani, please check the related issue for context information.
The aim here is to pass this header to every request. So, I've added it to normalize_headers() in request.py.

Please check request.py line 12. Is there a way to quickly retrieve the name and version of the current package in Python?
I am thinking of using the importlib-metadata package (https://pypi.org/project/importlib-metadata/). Do you think that this is a good approach or is there a better way?

### CODE REVIEW

File: arango/request.py [hunk 1]
```diff
@@ -2,13 +2,21 @@
 
 from typing import Any, MutableMapping, Optional
 
-from arango.typings import Fields, Headers, Params
+from arango.typings import DriverFlags, Fields, Headers, Params
 
 
-def normalize_headers(headers: Optional[Headers]) -> Headers:
+def normalize_headers(
+    headers: Optional[Headers], driverFlags: Optional[DriverFlags] = None
```
Comment (joowani) [inline arango/request.py:9]: nit: `driverFlags` -> `driver_flags` (we won't format the field names in JSON payload and response bodies to be snake_case anymore, but we should still use snake_case for the Python code itself).

File: arango/request.py [hunk 2]
```diff
@@ -2,13 +2,21 @@
...
+) -> Headers:
+    flags = ""
+    if driverFlags is not None:
+        for flag in driverFlags:
+            flags = flags + flag + ";"
+    driverInfo = "python-arango/7.5.2 (" + flags + ")"
```
Comment (joowani) [inline arango/request.py:15]: Same as above. `driverInfo` -> `driver_info` (or better yet `driver_header`).

File: tests/test_request.py [hunk 3]
```diff
@@ -11,15 +11,15 @@ def test_request_no_data():
...
-    assert request.headers == {
-        "charset": "utf-8",
-        "content-type": "application/json",
-        "foo": "bar",
-    }
+    # assert request.headers == {
+    #     "charset": "utf-8",
+    #     "content-type": "application/json",
+    #     "foo": "bar",
+    # }
```
Comment (joowani) [inline tests/test_request.py:18]: We can just change these to:
```
assert request.headers["charset"] == "utf-8"
assert request.headers["content-type"] == "application/json"
assert request.headers["foo"] == "bar" 
```

**Comment by joowani** [comment]
> Please check request.py line 12. Is there a way to quickly retrieve the name and version of the current package in Python? I am thinking of using the importlib-metadata package (https://pypi.org/project/importlib-metadata/). Do you think that this is a good approach or is there a better way?

You can simply do:
```
from arango.version import __version__
```
I think it's safe to assume that when we are executing the python-arango code, the version file would've already been generated by `pip install` (`pip install -e` in dev).

File: arango/request.py (hunk 1 above)
Comment (tjoubert) [reply arango/request.py:9]: This is fixed.

File: tests/test_request.py (hunk 3 above)
Comment (tjoubert) [reply tests/test_request.py:18]: This is fixed.

**Comment by tjoubert** [comment]
> > Please check request.py line 12. Is there a way to quickly retrieve the name and version of the current package in Python? I am thinking of using the importlib-metadata package (https://pypi.org/project/importlib-metadata/). Do you think that this is a good approach or is there a better way?
> 
> You can simply do:
> 
> ```
> from arango.version import __version__
> ```
> 
> I think it's safe to assume that when we are executing the python-arango code, the version file would've already been generated by `pip install` (`pip install -e` in dev).

@joowani , thanks for the tip. I tested it and it works!

File: arango/request.py (hunk 2 above)
Comment (tjoubert) [reply arango/request.py]: This is fixed

File: tests/test_request.py [hunk 4]
```diff
@@ -30,15 +30,15 @@ def test_request_string_data():
...
```
Comment (joowani) [inline tests/test_request.py:37]: Looks like you missed this one.

File: tests/test_request.py [hunk 5]
```diff
@@ -68,9 +66,9 @@ def test_request_transaction_data():
...
     assert request.params == {"bool": "1"}
-    assert request.headers == {
-        "charset": "utf-8",
-        "content-type": "application/json",
-        "foo": "bar",
-    }
+    # assert request.headers == {
+    #     "charset": "utf-8",
+    #     "content-type": "application/json",
+    #     "foo": "bar",
```
Comment (joowani) [inline tests/test_request.py:72]: This one also.

File: arango/request.py [hunk 6]
```diff
@@ -86,6 +99,7 @@ class Request:
         "write",
         "exclusive",
         "deserialize",
+        "driverFlags",
```
Comment (joowani) [inline arango/request.py:102]: driverFlags -> driver_flags

**Review by joowani — APPROVED** [review]
Looks like you missed a couple of spots but the rest looks good! Please feel free to merge after you address the nits. Thanks as always.

File: tests/test_request.py (hunk 4 above)
Comment (tjoubert) [reply tests/test_request.py:37]: Fixed

File: tests/test_request.py (hunk 5 above)
Comment (tjoubert) [reply tests/test_request.py:72]: Fixed

File: arango/request.py (hunk 6 above)
Comment (tjoubert) [reply arango/request.py:102]: Fixed

**Pull request merged**